buffer.go

```diff
@@ -0,0 +1,139 @@
+package kinesis
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"html/template"
+	"os"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/kinesis"
+	"github.com/gliderlabs/logspout/router"
+)
+
+// PutRecordsLimit is the maximum number of records allowed for a PutRecords request.
+var PutRecordsLimit = 500
```

Comment:
We flush every second here: https://github.com/remind101/logspout-kinesis/blob/mvp/drainer.go#L21, 500 is a kinesis hard limit.